src/main/java/org/roaringbitmap/BitSetUtil.java

```diff
@@ -1,138 +1,120 @@
+    
+    // a block consists has a maximum of 1024 words, each representing 64 bits,
+    // thus representing at maximum 65536 bits
+    static final private int BLOCK_LENGTH = BitmapContainer.MAX_CAPACITY / Long.SIZE; //
+                                                                      // 64-bit
+                                                                      // word
+
+    /**
+     * Compares a RoaringBitmap and a BitSet. They are equal
+     * if and only if they contain the same set of integers.
+     * 
+     * @param bitset first object to be compared
+     * @param bitmap second object to be compared
+     * @return
+     */
+    public static boolean equals(final BitSet bitset,
+            final RoaringBitmap bitmap) {
+        if(bitset.cardinality() != bitmap.getCardinality()) return false;
+        final IntIterator it = bitmap.getIntIterator();
+        while (it.hasNext()) {
+            int val = it.next();
+            if(!bitset.get(val)) return false;
+        }
+        return true;
+    }
+
+    
+    /**
+     * Generate a RoaringBitmap out of a BitSet
+     * 
+     * @param bitSet original bitset (will not be modified)
+     * @return roaring bitmap equivalent to BitSet
+     */
+    public static RoaringBitmap bitmapOf(final BitSet bitSet) {
+        return bitmapOf(bitSet.toLongArray());
+    }
 
-	/**
-	 * Generate RoaringBitmap out of the bitSet
-	 * 
-	 * @param bitSet
-	 * @return roaring bitmap
-	 */
-	static RoaringBitmap bitmapOf(final BitSet bitSet) {
-		final int cardinality = bitSet.cardinality();
-		if (cardinality == 0) {
-			return new RoaringBitmap();
-		} else if (cardinality <= ArrayContainer.DEFAULT_MAX_SIZE) {
-			// if it is a tiny bitSet, we can directly build a arrayContainer, without an extra bitSet#toLongArray
-			final RoaringBitmap bitmap = new RoaringBitmap();
-			bitmap.highLowContainer.append((short)0, arrayContainerOf(cardinality, bitSet));
-			return bitmap;
-		} else {
-			// before jdk 1.7
-			return bitmapOf(cardinality, toLongArray(cardinality, bitSet));
-			// starting with jdk.17
-			//return bitmapOf(cardinality, bitSet.toLongArray());
-		}
-	}
-	
-	private static long[] toLongArray(final int cardinality, final BitSet bitSet) {		
-		final long[] words = new long[(bitSet.length() + 63) / 64];
-	    for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i+1)) {
-            if (i == Integer.MAX_VALUE) {
-                break;
+    /**
+     * Generate a RoaringBitmap out of a long[], each long using little-endian
+     * representation of its bits
+     * 
+     * @see BitSet#toLongArray() for an equivalent
+     * @param words array of longs  (will not be modified)
+     * @return roaring bitmap
+     */
+    public static RoaringBitmap bitmapOf(final long[] words) {
+        // split long[] into blocks.
+        // each block becomes a single container, if any bit is set
+        final RoaringBitmap ans = new RoaringBitmap();
```

Comment:
And do you think, we should have a check here for long[] not to exceed the maximum of representable bits in Roaring? Or limiting word.length in general....and ignoring the rest of the long[] ?